diff --git a/pdf_viewer/document_view.cpp b/pdf_viewer/document_view.cpp
index 3438fabd..8446aec7 100644
--- a/pdf_viewer/document_view.cpp
+++ b/pdf_viewer/document_view.cpp
@@ -909,7 +913,7 @@ void DocumentView::fit_to_page_height(bool smart) {
         imbalance = 0;
     }
 
-    set_zoom_level(static_cast<float>(view_height - 20) / page_height, true);
+    set_zoom_level(static_cast<float>(view_height + 2) / page_height, true);
     //set_offset_y(-imbalance * normal_page_height / 2.0f);
     goto_offset_within_page(cp, (imbalance / 2.0f + 0.5f) * normal_page_height);
 }
