diff --git a/pdf_viewer/document_view.cpp b/pdf_viewer/document_view.cpp
index 3438fabd..41938348 100644
--- a/pdf_viewer/document_view.cpp
+++ b/pdf_viewer/document_view.cpp
@@ -744,6 +744,10 @@ void DocumentView::move_pages(int num_pages) {
     int padding = two_page_mode ? page_space_y : PAGE_PADDINGS;
 
     move_virtual(0, num_pages * (current_document->get_page_height(current_page) + padding));
+
+    if (fast_coordinates()) {
+        set_offsets(get_offset_x(), get_offset_y());
+    }
 }
 
 void DocumentView::move_screens(int num_screens) {
