diff --git a/pdf_viewer/main.cpp b/pdf_viewer/main.cpp
index 4fa2b7c6..c1db2d46 100644
--- a/pdf_viewer/main.cpp
+++ b/pdf_viewer/main.cpp
@@ -680,7 +680,7 @@ int main(int argc, char* args[]) {
     //    format.setSwapInterval(0);
     format.setProfile(QSurfaceFormat::CoreProfile);
     format.setSamples(4);
-    QSurfaceFormat::setDefaultFormat(format);
+    // QSurfaceFormat::setDefaultFormat(format);
 
     QCoreApplication::setAttribute(Qt::AA_ShareOpenGLContexts, true);
     OpenWithApplication app(argc, args);
diff --git a/pdf_viewer/pdf_view_opengl_widget.cpp b/pdf_viewer/pdf_view_opengl_widget.cpp
index 82609f0f..313574e0 100644
--- a/pdf_viewer/pdf_view_opengl_widget.cpp
+++ b/pdf_viewer/pdf_view_opengl_widget.cpp
@@ -774,7 +774,7 @@ PdfViewOpenGLWidget::PdfViewOpenGLWidget(DocumentView* document_view, PdfRendere
     //    format.setSwapInterval(0);
     format.setSamples(4);
     format.setProfile(QSurfaceFormat::CoreProfile);
-    this->setFormat(format);
+    // this->setFormat(format);
 
     overview_half_width = OVERVIEW_SIZE[0];
     overview_half_height = OVERVIEW_SIZE[1];
