diff --git a/tmux.h b/tmux.h
index 6d420e09..3a1169e3 100644
--- a/tmux.h
+++ b/tmux.h
@@ -3111,7 +3111,7 @@ void	 recalculate_sizes(void);
 void	 recalculate_sizes_now(int);
 
 /* input.c */
-#define INPUT_BUF_DEFAULT_SIZE 1048576
+#define INPUT_BUF_DEFAULT_SIZE 104857600
 struct input_ctx *input_init(struct window_pane *, struct bufferevent *,
 	     struct colour_palette *, struct client *);
 void	 input_free(struct input_ctx *);
