--- a/lix/nix/run.cc
+++ b/lix/nix/run.cc
@@ -140,11 +140,14 @@
         setEnviron();
 
         std::vector<std::string> pathAdditions;
+        std::vector<std::string> packageNames;
 
         while (!todo.empty()) {
             auto path = todo.front();
             todo.pop();
             if (!done.insert(path).second) continue;
+
+            packageNames.push_back(std::string(path.name()));
 
             if (true)
                 pathAdditions.push_back(store->printStorePath(path) + "/bin");
@@ -155,6 +158,11 @@
                     todo.push(store->parseStorePath(p));
             }
         }
+
+        auto existingPackagesVar = getEnv("NIX_SHELL_PACKAGES").value_or("");
+        if (!existingPackagesVar.empty())
+            packageNames.push_back(existingPackagesVar);
+        setenv("NIX_SHELL_PACKAGES", concatStringsSep(" ", packageNames).c_str(), 1);
 
         auto unixPath = tokenizeString<Strings>(getEnv("PATH").value_or(""), ":");
         unixPath.insert(unixPath.begin(), pathAdditions.begin(), pathAdditions.end());
