diff --git a/env.cc b/env.cc
index d91ee72d7..bb980bddd 100644
--- a/env.cc
+++ b/env.cc
@@ -1,5 +1,6 @@
 #include <unordered_set>
 #include <queue>
+#include <boost/algorithm/string/join.hpp>
 
 #include "nix/cmd/command.hh"
 #include "nix/expr/eval.hh"
@@ -76,9 +77,8 @@ struct CmdShell : InstallablesCommand, MixEnvironment
         for (auto & path : outPaths)
             todo.push(path);
 
-        setEnviron();
-
         std::vector<std::string> pathAdditions;
+        std::vector<std::string> packageNames;
 
         while (!todo.empty()) {
             auto path = todo.front();
@@ -98,8 +98,17 @@ struct CmdShell : InstallablesCommand, MixEnvironment
                 for (auto & p : tokenizeString<Paths>(state->storeFS->readFile(propPath)))
                     todo.push(store->parseStorePath(p));
             }
+
+            packageNames.push_back(std::string(path.name()));
         }
 
+        auto existingPackagesVar = getEnvOs(OS_STR("NIX_SHELL_PACKAGES")).value_or(OS_STR(""));
+        if (!existingPackagesVar.empty())
+            packageNames.push_back(existingPackagesVar);
+        setVars["NIX_SHELL_PACKAGES"] = boost::algorithm::join(packageNames, " ");
+
+        setEnviron();
+
         // TODO: split losslessly; empty means .
         auto unixPath = ExecutablePath::load();
         unixPath.directories.insert(unixPath.directories.begin(), pathAdditions.begin(), pathAdditions.end());
