diff --git a/src/animation/client.h b/src/animation/client.h
index bee8bff..7c520dd 100644
--- a/src/animation/client.h
+++ b/src/animation/client.h
@@ -637,11 +637,11 @@ void client_draw_border(Client *c, struct ivec2 offsets) {
 		return;
 
 	if (c->isfullscreen) {
+		wlr_scene_node_set_position(&c->scene_surface->node, 0, 0);
 		if (c->border->node.enabled) {
 			wlr_scene_node_set_enabled(&c->splitindicator[0]->node, false);
 			wlr_scene_node_set_enabled(&c->splitindicator[1]->node, false);
 			wlr_scene_node_set_enabled(&c->border->node, false);
-			wlr_scene_node_set_position(&c->scene_surface->node, 0, 0);
 		}
 		return;
 	} else {
diff --git a/src/layout/arrange.h b/src/layout/arrange.h
index 5fa9031..e947c9c 100644
--- a/src/layout/arrange.h
+++ b/src/layout/arrange.h
@@ -1238,6 +1238,13 @@ arrange(Monitor *m, bool want_animation, bool from_view) {
 
 	pre_calculate_before_arrange(m, want_animation, from_view, false);
 
+	Client *_c;
+	wl_list_for_each(_c, &clients, link) {
+		if (!VISIBLEON(_c, m) || !ISTILED(_c))
+			continue;
+		wlr_scene_node_set_enabled(&_c->border->node, 1);
+	}
+
 	if (m->isoverview) {
 		overviewlayout.arrange(m);
 	} else {
diff --git a/src/layout/horizontal.h b/src/layout/horizontal.h
index f3fd0fa..dfb97e8 100644
--- a/src/layout/horizontal.h
+++ b/src/layout/horizontal.h
@@ -602,10 +602,11 @@ monocle(Monitor *m) {
 	wl_list_for_each(c, &clients, link) {
 		if (!VISIBLEON(c, m) || !ISFAKETILED(c))
 			continue;
-		geom.x = m->w.x + cur_gappoh;
-		geom.y = m->w.y + cur_gappov;
-		geom.width = m->w.width - 2 * cur_gappoh;
-		geom.height = m->w.height - 2 * cur_gappov;
+		wlr_scene_node_set_enabled(&c->border->node, 0);
+		geom.x = m->w.x + cur_gappoh - c->bw;
+		geom.y = m->w.y + cur_gappov - c->bw;
+		geom.width = m->w.width - 2 * cur_gappoh + 2 * c->bw;
+		geom.height = m->w.height - 2 * cur_gappov + 2 * c->bw;
 		client_tile_resize(c, geom, 0);
 	}
 }
